openpype/hosts/tvpaint/plugins/publish/extract_sequence.py

```diff
@@ -434,27 +599,153 @@ def _render_layer(
+            if match:
+                files.append(match.group())
+                reference_file_pattern = filepath
+
+        if len(files) == 1:
+            # Prepare which hole is filled with what frame
+            #   - the frame is filled only with already existing frames
+            prev_frame = files[0]
+            match = re.search(r'0*(\d+)\.png', prev_frame)
+            if match:
+                index = [int(match.group(1))]
+
+        collections = clique.assemble(files)[0]
+        if len(collections) != 1 and not index:
+            raise KnownPublishError(
+                "Multiple collections {} found.".format(collections))
+
+        if not index:
+            col = collections[0]
+            index = col.indexes
+            # Prepare which hole is filled with what frame
+            #   - the frame is filled only with already existing frames
+            prev_frame = next(iter(col.indexes))
+
+        hole_frame_to_nearest = {}
+        for frame in range(int(start_frame), int(end_frame) + 1):
+            if frame in index:
+                prev_frame = frame
+
+            elif frame in frame_to_avoid:
+                continue
+
+            else:
+                # Use previous frame as source for hole
+                hole_frame_to_nearest[frame] = prev_frame
+
+        # Calculate paths
+        added_filepaths_by_frame_index = {}
+        finale_template = get_frame_filename_template(end_frame)
+
+        #col_format = col.format("{head}{padding}{tail}")
+        for hole_frame, src_frame in hole_frame_to_nearest.items():
+            hole_f = finale_template.format(frame=hole_frame)
+            src_f = finale_template.format(frame=src_frame)
+            hole_fpath = re.sub(r'0*(\d+)\.png', hole_f, reference_file_pattern)
+            src_fpath = re.sub(r'0*(\d+)\.png', src_f, reference_file_pattern)
+            if not os.path.isfile(src_fpath):
+                raise KnownPublishError(
+                    "Missing previously detected file: {}".format(src_fpath))
+
+            speedcopy.copyfile(src_fpath, hole_fpath)
+            added_filepaths_by_frame_index[hole_frame] = hole_fpath
+
+        return added_filepaths_by_frame_index
+
+    def _create_bg_images(self, bg_layer, output_dir, resolution, custom_mark_range):
+
+        bg_color = self._get_review_bg_color(review=False)
+
+        filepaths_by_frame = {}
+
+        layer_template = get_layer_pos_filename_template(range_end=bg_layer["frame_end"])
+
+        for frame_idx in range(bg_layer["frame_start"], bg_layer["frame_end"] + 1):
+            filename = layer_template.format(pos=bg_layer["position"],frame=frame_idx)
+            dst_path = "/".join([output_dir, filename])
+            filepaths_by_frame[frame_idx] = dst_path
+
+            bg_image = Image.new("RGBA", resolution, bg_color)
+            bg_image.save(dst_path, "PNG")
 
         return filepaths_by_frame
+
+    def _convert_opacity_range(self, value):
```

Comment:
Why not simply do:
```python
return max(0, min(int(round(value * 2.55)), 255))
```
Where `2.55` is the factor, `0` the min value and `255` the max one.

Reply:
Because I didn't thought of it that way ^^'

Reply:
If you agree, I'll keep the tip for further usage, it's cool to put it in one line !